src/librawspeed/decompressors/CrxDecompressor.cpp

```diff
@@ -1625,9 +1625,9 @@ static int crxParamInit(CrxImage* img, CrxBandParam** param,
 
 static int crxSetupSubbandData(CrxImage* img, CrxPlaneComp* planeComp,
                         const CrxTile* tile, uint32_t mdatOffset) {
-  long compDataSize = 0;
-  long waveletDataOffset = 0;
-  long compCoeffDataOffset = 0;
+  int64_t compDataSize = 0;
```

Comment:
You have replaced **short** with int16_t on other locations, shouldn't be long mapped to int32_t instead?

Reply:
I replaced with what LLVM suggested in https://github.com/darktable-org/rawspeed/pull/271/checks?check_run_id=2639080878

eg: `warning: consider replacing 'unsigned long' with 'uint64_t' [google-runtime-int]`
or: `warning: consider replacing 'long' with 'int64_t' [google-runtime-int]`